### ISSUE TITLE
Character references in link definition labels

### ISSUE THREAD

**Issue by wooorm** [commonmark/commonmark-spec#616]
- Character references are allowed everywhere, except in fenced code, indented code, or code spans
- They represent their resolved character, not syntax

There’s even [example 318](https://spec.commonmark.org/0.29/#example-318) of having them in link definition destinations and link definition titles.

But, [the following does not resolve into a link](https://spec.commonmark.org/dingus/?text=%5B%26copy%3B%5D%3A%20example.com%0A%0A%5B©%5D%0A):

```markdown
[&copy;]: example.com

[©][]
```

---

I interpret the spec as saying that it should resolve, but then the dingus doesn’t.
This may be a bug for the dingus implementation, rather than the spec.

**Comment by Crissov** [comment]
I agree that, to meet author expectation or intuition, character references of all kinds should be normalized in link labels (and elsewhere), especially since letter case is being ignored. Unfortunately, only a single implementation, Maruku, does it this way, although most CM-conformant parsers (and Pandoc) will happily convert any HTML entities to plain characters on output.

**Comment by mgeier** [comment]
This might be related to #572.

**Comment by wooorm** [comment]
Btw, I think this should be true for character escapes too:

```markdown
[&copy;]: a.com
[\!]: b.com

Both should link: [©], [!]
```

Yields:

[&copy;]: a.com
[\!]: b.com

Both should link: [©], [!]

**Comment by wooorm** [comment]
@jgm Is this something you agree with? I can create a PR to clarify the docs

**Comment by vassudanagunta** [comment]
I don't think this needs clarification of the docs so much as a bug report against CommonMark.js. 

That said, [every single Markdown implementation but one fails this test](https://johnmacfarlane.net/babelmark2/?normalize=1&text=%5B%26copy%3B%5D%3A+example.com%0A%0A%5B©%5D%5B%5D).
